sycl/test-e2e/Matrix/joint_matrix_apply_bf16_impl.hpp

```diff
@@ -36,41 +39,74 @@ void matrix_verify_add(big_matrix<T, M, N> &A, const float ref, F &&lambda) {
            const auto sg_starty = global_idy - spmd_item.get_local_id(1);
 
            sub_group sg = spmd_item.get_sub_group();
-           joint_matrix<sub_group, T, use::a, TM, TK, layout::row_major> sub_a;
+           joint_matrix<sub_group, T, use::a, TileRows, TileCols,
+                        layout::row_major>
+               sub_a;
 
-           joint_matrix_fill(sg, sub_a, bfloat16(5.0));
+           joint_matrix_fill(sg, sub_a, T(5.0));
 
            joint_matrix_apply(sg, sub_a, lambda);
 
            ext::intel::experimental::matrix::joint_matrix_store(
                sg, sub_a,
                accA.template get_multi_ptr<access::decorated::no>() +
-                   (sg_startx * TM) * N + sg_starty / sg_size * TN,
-               N);
+                   (sg_startx * TileRows) * Cols +
+                   sg_starty / sg_size * TileCols,
+               Cols);
          }); // parallel for
    }).wait();
   // Check if the results are correct
   {
     host_accessor Acc{bufA};
     assert(std::all_of(Acc.begin(), Acc.end(), [=](auto Elem) {
-      return (std::fabs(static_cast<float>(make_fp32(Elem) - ref)) <
-              std::numeric_limits<float>::epsilon());
+      return (std::fabs(static_cast<TResult>(make_fp32(Elem) - ref)) <
+              std::numeric_limits<TResult>::epsilon());
     }));
   }
 }
 
-static constexpr size_t MATRIX_M = TM * 2;
-static constexpr size_t MATRIX_N = TN * 2;
-bfloat16 A[MATRIX_M][MATRIX_N];
+template <typename T, typename TResult, size_t TM, size_t TK> void test() {
```

Comment:
Since several combinations are being tested in this test, could you add output, which combination is being tested, so that when assert happens we can see from the output right away, which combination is broken?